application/application.go

```
@@ -23,15 +30,44 @@ func New(application, service string) Tags {
 }
 
 // Map with all values
-func (a Tags) Map() map[string]string {
+func (app *Tags) Map() map[string]string {
+	os.Getenv("pp")
 	allTags := make(map[string]string)
-	allTags["application"] = a.Application
-	allTags["service"] = a.Service
-	allTags["cluster"] = a.Cluster
-	allTags["shard"] = a.Shard
+	allTags["application"] = app.Application
+	allTags["service"] = app.Service
+	allTags["cluster"] = app.Cluster
+	allTags["shard"] = app.Shard
 
-	for k, v := range a.CustomTags {
+	for k, v := range app.CustomTags {
 		allTags[k] = v
 	}
 	return allTags
 }
+
+// AddCustomTagsFromEnv set additional custom tags from environment variables that match the given regex.
+func (app *Tags) AddCustomTagsFromEnv(regx string) {
+	r, err := regexp.Compile(regx)
+	if err != nil {
+		log.Printf("Error creating custom tags: %v\n", err)
```

Return an error here rather than just log it?